packages/ember-glimmer/lib/utils/iterable.js

```diff
@@ -175,6 +175,12 @@ class Iterable {
       return new EmberArrayIterator(iterable, keyFor);
     } else if (Array.isArray(iterable)) {
       return iterable.length > 0 ? new ArrayIterator(iterable, keyFor) : EMPTY_ITERATOR;
+    } else if (typeof iterable.forEach === 'function') {
+      let array = [];
+      iterable.forEach(function(item) {
+        array.push(item);
+      });
+      return array.length > 0 ? new ArrayIterator(array, keyFor) : EMPTY_ITERATOR;
     } else if (isEachIn(ref)) {
```

Comment:
isEachIn probably needs to be moved up – if you do `{{#each-in someArray ...}}` for some reason, it should do what you asked for instead of looping through the array. It's largely unrelated to what you are doing here because even if you move it above your branch it will still be a problem, but we should remember to fix it and add a test.